functions/user.js

```diff
@@ -0,0 +1,82 @@
+const config = require("./config");
+const GitHubbub = require('./githubbub');
+const util = require("util");
+const Promise = require('bluebird');
+const _ = require("lodash");
+
+function UserWrapper(admin) {
+
+  class User {
+    static findById(id) {
+      let db = admin.database();
+      return Promise.props({
+        account: db.ref(`/accounts/${id}`).once("value").then(snapshot => {
+          return snapshot.val();
+        }),
+        profile: db.ref(`/profiles/${id}`).once("value").then(snapshot => {
+          return snapshot.val();
+        })
+      }).then(data => {
+        let
+          account = data.account || {}
+        , profile = data.profile || {};
+
+        return new User({
+          id: id,
+          email: account.email,
+          githubToken: account.githubToken,
+          updatedAt: account.updatedAt,
+          githubCreatedAt: account.githubCreatedAt,
+          name: profile.name,
+          handle: profile.handle,
+          photo: profile.photo
+        });
+      });
+    }
+
+    constructor (attributes) {
+      this.id = attributes.id;
+      this.email = attributes.email;
+      this.githubToken = attributes.githubToken;
+      this.updatedAt = attributes.updatedAt;
+      this.githubCreatedAt = attributes.githubCreatedAt;
+      this.name = attributes.name;
+      this.handle = attributes.handle;
+      this.photo = attributes.photo;
+      this.uid = attributes.uid;
+      this.db = admin.database();
+      this.github = new GitHubbub(this.githubToken);
+    }
+
+    updateProfile () {
```

Comment:
It seems weird to fetch the user just to make this an instance method... maybe it should be a class method and just pass in the ID?  Seems like we're doing two reads of the user record (one before this call and one at the end) which we don't actually need.  Confused why that's needed.